public/src/main/java/com/revenuecat/purchases/models/GoogleProrationMode.kt

```diff
@@ -0,0 +1,28 @@
+package com.revenuecat.purchases.models
+
+import com.android.billingclient.api.BillingFlowParams
+
+/**
+ * Enum of possible proration modes to be passed to a Google Play purchase.
+ * Ignored for Amazon purchases. <-- TODOBC5 confirm
+ *
+ * See https://developer.android.com/google/play/billing/subscriptions#proration for examples
+ */
+enum class GoogleProrationMode(@BillingFlowParams.ProrationMode val playBillingClientName: Int) {
+    /**
+     * New subscription benefits take effect immediately.
+     * User is charged for the full price of the new subscription on the old subscription's expiration date.
+     *
+     * This is the default behavior.
+     */
+    IMMEDIATE_WITHOUT_PRORATION(BillingFlowParams.ProrationMode.IMMEDIATE_WITHOUT_PRORATION),
+
+    /**
+     * New subscription benefits take effect immediately.
+     * Any time remaining on the old subscription is used to push out the first payment date for the new subscription.
```

Comment:
another potential phrasing:

`Waits x number of days to charge the full price of the new subscription, where x represents the number of days that has already been paid for by previous subscription credit`

seems a little wordy, though a bit more specific...

Reply:
Hmm I think both work but I kinda prefer what you have... It could also be good to mention how this would look in our data infrastructure. As in, the old subscription canceled immediately and the new one started (I believe that's how it would look like but we should check). Something like:
`The old subscription is canceled immediately and the remaining time is proportionally added to the new subscription according to the new subscription value`

Reply:
ah, like something about how entitlements work? good call